data/transactions/transaction.go

```diff
@@ -227,7 +227,13 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 			return err
 		}
 	case protocol.KeyRegistrationTx:
-		// All OK
+		// check that, if this tx is marking an account nonparticipating, it supplies no key (as though it were trying to go offline)
+		if tx.KeyregTxnFields.Nonparticipation {
+			suppliesNullKeys := tx.KeyregTxnFields.VotePK == crypto.OneTimeSignatureVerifier{} || tx.KeyregTxnFields.SelectionPK == crypto.VRFVerifier{}
```

Comment:
Shouldn't this be `&&`?

Reply:
The idea is to mimic the logic in `KeyregTxnFields.apply`, where just having one of the two zeroed-out is sufficient - see `keyreg.go:59` or so.

Reply:
Ah, I see.  Sure.